src/components/video/VideoJS.jsx

```diff
@@ -0,0 +1,72 @@
+import React, { useEffect, useRef } from 'react';
+import PropTypes from 'prop-types';
+
+import 'videojs-youtube';
+import videojs from 'video.js';
+import 'video.js/dist/video-js.css';
+
+const VideoJS = ({ options, onReady }) => {
+  const videoRef = useRef(null);
+  const playerRef = useRef(null);
+
+  useEffect(() => {
+    // Make sure Video.js player is only initialized once
+    if (!playerRef.current) {
+      // The Video.js player needs to be _inside_ the component el for React 18 Strict Mode.
+      const videoElement = document.createElement('video-js');
+
+      videoElement.classList.add('vjs-big-play-centered');
+      videoRef.current.appendChild(videoElement);
+
+      // eslint-disable-next-line no-multi-assign
+      const player = playerRef.current = videojs(videoElement, options, () => {
+        if (onReady) {
+          onReady(player);
```

Comment:
I tried adding tests for it but I was not able to find the right implementation that could wait for the player to start playing inside jest. I do not know if it is possible or not. I have tested this part manually though and it is working correctly. I am skipping tests since this part is provided by video.js and it is very likely that they have tests in place on their side.